lib/trie/lookup.go

```diff
@@ -18,24 +18,24 @@ type recorder interface {
 
 // findAndRecord search for a desired key recording all the nodes in the path including the desired node
 func findAndRecord(t *Trie, key []byte, recorder recorder) error {
-	return find(t.root, key, recorder)
+	return find(t.root, key, recorder, true)
```

Comment:
nit for extreme readability

```suggestion
	const isRoot = true
	return find(t.root, key, recorder, isRoot)
```